zpeclient.go

```diff
@@ -293,10 +296,22 @@ func (c *Cache) authorize(principal string, check AthenzAccessCheck) (bool, erro
 	for _, role := range roles {
 		policies := domainData.roleToAssertion[role]
 		for _, assert := range policies {
-			if assert.resource.MatchString(check.Resource) && assert.action.MatchString(check.Action) && *assert.effect == zms.ALLOW {
-				return true, nil
+			effect := *assert.effect
+			if effect == 0 {
+				effect = zms.ALLOW
```

Comment:
looks like we are not using `effect` further below? why do we have to assign to `ALLOW` by default?

Reply:
fixed.